src/psyclone/tests/psyir/backend/sympy_writer_test.py

```diff
@@ -197,15 +231,42 @@ def test_sym_writer_symbol_types(fortran_reader, expressions):
     '''
     # A dummy program to easily create the PSyIR for the
     # expressions we need. We just take the RHS of the assignments
-    source = '''program test_prog
+    source = f'''program test_prog
                 use my_mod
                 type(my_type) :: a, b(10)
                 integer :: i, j, x, a_c, b_c, a_b_c, a_b_c_1
-                x = {0}
-                end program test_prog '''.format(expressions[0])
+                x = {expressions[0]}
+                end program test_prog '''
 
     psyir = fortran_reader.psyir_from_source(source)
     expr = psyir.children[0].children[0].rhs
-    sympy_writer = SymPyWriter(list_of_expressions=[expr])
+    type_map = SymPyWriter.create_type_map([expr])
+    sympy_writer = SymPyWriter(type_map)
+    # Note that this call can extend the type_map with type information
+    # about member names.
     _ = sympy_writer(expr)
-    assert sympy_writer.get_sympy_types() == expressions[1]
+    assert sympy_writer._sympy_type_map == expressions[1]
+
+
+def test_sym_writer_convert_to_sympy_expressions(fortran_reader):
+    '''Tests that convenience function `write_as_sympy_strings works
```

Comment:
Should be `convert_to_sympy_expressions` ?

Reply:
Yes.